api/v1alpha1/nodemaintenance_types.go

```diff
@@ -42,6 +48,12 @@ const (
 	ConditionReasonReady string = "Ready"
 	// ConditionReasonAborted means that NodeMaintenance is in Aborted state
 	ConditionReasonAborted string = "Aborted"
+
+	// Condition Reasons for ConditionTypeFailed condition type.
+
+	// ConditionReasonFatal means that maintenance operation failed in a non-recoverable way
+	// garbage collection will not occur and NodeMaintenance will remain in Ready state.
+	ConditionReasonFatal string = "Fatal"
```

Comment:
Do we really need the new condition? It is not clear for me how the caller should behave. It is expected that the caller will check both conditions (Ready and Failed)?

Reply:
See details in https://github.com/Mellanox/maintenance-operator/pull/12#discussion_r1727409435